pkg/virt-api/rest/subresource.go

```diff
@@ -598,13 +598,19 @@ func (app *SubresourceAPIApp) StopVMRequestHandler(request *restful.Request, res
 		if !errors.IsNotFound(err) {
 			writeError(errors.NewInternalError(err), response)
 			return
-		} else {
-			writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("VM is not running")), response)
-			return
 		}
+
+		writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("VM has no associated VMI running")), response)
+		return
 	}
-	if vmi == nil || vmi.IsFinal() || vmi.Status.Phase == v1.Unknown || vmi.Status.Phase == v1.VmPhaseUnset {
-		writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("VM is not running")), response)
+
+	if vmi == nil || vmi.Status.Phase == v1.Succeeded || vmi.Status.Phase == v1.Unknown || vmi.Status.Phase == v1.VmPhaseUnset {
+		writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("VM has status %q and is not running", vmi.Status.Phase)), response)
+		return
+	}
+
+	if vmi.Status.Phase == v1.Failed && bodyStruct.GracePeriod == nil {
+		writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("VM is Failed. Use `--force --gracePeriod=0` to stop the VM")), response)
```

Comment:
I'm probably missing something here - but I don't understand where the code addresses `--force`, I only see 
`--gracePeriod` (`bodyStruct.GracePeriod`) being addressed. Can you elaborate?

Reply:
yup, if you look at the `stop` command in order to set gracePeriod to 0 you have to use the `--force` command as well as the `--force` can not be used without `--gracePeriod 0`. This was for consistency with the restart command which has the same flow. Both flags are needed for the force stop.

Reply:
It also seems that this only verifies that `--grace-period` is set to _something_ (not necessarily 0). Should we also enforce that `*bodyStruct.GracePeriod == 0`?

Reply:
ah yeah, let's do that. Right now it can only be set to 0, but makes sense to future proof it. thanks!